Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+
+    uint64 TransferCOunt = 2;
+}
+
+message ReportedProcess_XLGpp{
+    string Path = 1;
+
+    uint32 ProcessId = 2;
+
+    string ProcessArgs = 3;
+
+    IOTypeCounters_XLGpp ReadCounters = 4;
+
+    IOTypeCounters_XLGpp WriteCounters = 5;
+
+    IOTypeCounters_XLGpp OtherCounters = 6;
+
+    google.protobuf.Timestamp CreationTime = 7;
+
+    google.protobuf.Timestamp ExitTime = 8;
+
+    google.protobuf.Duration KernelTime = 9;
+
+    google.protobuf.Duration UserTime = 10;
+
+    uint32 ExitCode = 11;
+
+    uint32 ParentProcessId = 12;
+}
+
+message ReportedFileAccess_XLGpp{
+    CreationDisposition_XLGpp CreationDisposition = 1;
+
+    DesiredAccess_XLGpp DesiredAccess = 2;
+
+    uint32 Error = 3;
+
+    uint64 Usn = 4;
+
+    FlagsAndAttributes_XLGpp FlagsAndAttributes = 5;
+
+    string Path = 6;
+
+    string ManifestPath = 7;
+
+    ReportedProcess_XLGpp Process = 8;
+
+    ShareMode_XLGpp ShareMode = 9;
+
+    FileAccessStatus_XLGpp Status = 10;
+
+    FileAccessStatusMethod_XLGpp Method = 11;
+
+    RequestedAccess_XLGpp RequestedAccess = 12;
+
+    ReportedFileOperation_XLGpp Operation = 13;
+
+    bool ExplicitlyReported = 14;
+
+    string EnumeratePattern = 15;
+}
+
+message ProcessDetouringStatusData_XLGpp{
+    uint64 ProcessID = 1;
+
+    uint32 ReportStatus = 2;
+
+    string ProcessName = 3;
+
+    string StartApplicationName = 4;
+
+    string StartCommandLine = 5;
+
+    bool NeedsInjection = 6;
+
+    uint64 Job = 7;
+
+    bool DisableDetours = 8;
+
+    uint32 CreationFlags = 9;
+
+    bool Detoured = 10;
+
+    uint32 Error = 11;
+}
+
+message WeakContentFingerPrint_XLGpp{
+    Fingerprint_XLGpp Hash = 1;
+}
+
+message StrongContentFingerPrint_XLGpp{
+    Fingerprint_XLGpp Hash = 1;
+}
+
+message ObservedPathEntry_XLGpp{
+    AbsolutePath_XLGpp Path = 1;
+
+    string EnumeratePatternRegex = 2;
+}
+
+message IUnsafeSandboxConfiguration_XLGpp{
+    PreserveOutputsMode_XLGpp PreserveOutputs = 1;
+
+    bool MonitorFileAccesses = 2;
+
+    bool IgnoreZwRenameFileInformation = 3;
+
+    bool IgnoreZwOtherFileInformation = 4;
+
+    bool IgnoreNonCreateFileReparsePoints = 5;
+
+    bool IgnoreSetFileInformationByHandle = 6;
+
+    bool IgnoreReparsePoints = 7;
+
+    bool IgnorePreloadedDlls = 8;
+
+    bool ExistingDirectoryProbesAsEnumerations = 9;
+
+    bool MonitorNtCreateFile = 10;
+
+    bool MonitorZwCreateOpenQueryFile = 11;
+
+    SandboxKind_XLGpp SandboxKind = 12;
+
+    bool UnexpectedFileAccessesAreErrors = 13;
+
+    bool IgnoreGetFinalPathNameByHandle = 14;
+
+    bool IgnoreDynamicWritesOnAbsentProbes = 15;
+
+    DoubleWritePolicy_XLGpp DoubleWritePolicy = 16;
+
+    bool IgnoreUndeclaredAccessesUnderSharedOpaques = 17;
+}
+
+message UnsafeOptions_XLGpp{
+    ContentHash_XLGpp PreserveOutputsSalt = 1;
 
-    // Identifies the worker which invoked this event
-    uint32 WorkerID = 2;
+    IUnsafeSandboxConfiguration_XLGpp UnsafeConfiguration = 2;
+}
+
+message StringId_XLGpp{

```

Comment:
>StringId_XLGpp [](start = 8, length = 14)

why do you need this? StringID is an int. Wrapping it in a struct will only cost us space, no? What are the benefits?

Reply:
Benefit is that we know exactly what type it is, and if we were to add something else to the accompanying BXL "StringId" class, we can just add another field to this struct. 

---
In reply to: [308896523](https://github.com/microsoft/BuildXL/pull/671#discussion_r308896523) [](ancestors = 308896523)